src/main/java/de/tum/in/www1/artemis/domain/BaseExercise.java

```diff
@@ -213,8 +214,12 @@ public void validateDates() {
         if (isExamExercise()) {
             throw new BadRequestAlertException("An exam exercise may not have any dates set!", getTitle(), "invalidDatesForExamExercise");
         }
-        // at least one is set, so we have to check the two possible errors
-        boolean areDatesValid = isBeforeAndNotNull(getReleaseDate(), getDueDate()) && isValidAssessmentDueDate(getReleaseDate(), getDueDate(), getAssessmentDueDate());
+
+        IncludedInOverallScore includedInOverallScore = this instanceof Exercise thisAsExercise ? thisAsExercise.getIncludedInOverallScore() : null;
```

Comment:
I don't like this change. `BaseExercise` is a superclass of exercise, this line turns around the relation.

A better solution might be to make `isValidExampleSolutionPublicationDate` non-static and abstract / overwrite the implementation in the subclass to respect the `includedInOverallScore`-value here.

Reply:
Yes I wasn't comfortable with that line very much either. That sounds better thank you. What do you think about moving `validateDates` to `Exercise.java` and making `isValidExampleSolutionPublicationDate` etc. `protected` instead?

Reply:
I pushed the changes I mentioned for review but if you think the way you mentioned would be better I can change it.